tests/python_package_test/test_engine.py

```diff
@@ -3566,6 +3566,156 @@ def test_force_split_with_feature_fraction(tmp_path):
         assert tree_structure['split_feature'] == 0
 
 
+def test_goss_boosting_and_strategy_equivalent():
+    X, y = make_regression(n_samples=10_000, n_features=10, n_informative=5, random_state=42)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    lgb_train = lgb.Dataset(X_train, y_train)
+    lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train)
+    params1 = {
+        'boosting': 'goss',
+        'metric': 'l2',
+        'verbose': -1,
+        'bagging_seed': 0,
+        'learning_rate': 0.05,
+        'num_threads': 1,
+        'force_row_wise': True,
+        'gpu_use_dp': True,
+    }
+    evals_result1 = {}
+    lgb.train(params1, lgb_train,
+              num_boost_round=10,
+              valid_sets=lgb_eval,
+              callbacks=[lgb.record_evaluation(evals_result1)])
+    params2 = {
+        'data_sample_strategy': 'goss',
+        'metric': 'l2',
+        'verbose': -1,
+        'bagging_seed': 0,
+        'learning_rate': 0.05,
+        'num_threads': 1,
+        'force_row_wise': True,
+        'gpu_use_dp': True,
+    }
```

Comment:
```suggestion
    base_params = {
        'metric': 'l2',
        'verbose': -1,
        'bagging_seed': 0,
        'learning_rate': 0.05,
        'num_threads': 1,
        'force_row_wise': True,
        'gpu_use_dp': True,
    }
    params1 = {**base_params, 'boosting': 'goss'}
    evals_result1 = {}
    lgb.train(params1, lgb_train,
              num_boost_round=10,
              valid_sets=lgb_eval,
              callbacks=[lgb.record_evaluation(evals_result1)])
    params2 = {**base_params, 'data_sample_strategy': 'goss'}
```

It was really difficult for me to tell the difference between these two cases, because so many parameter values with identical values are repeated.

To make it clearer what's changing and to reduce the risk of accidental differences unrelated to sample strategy / boosting, will you please consider this consolidation? And the same for the other tests in this file.